src/kernels/MIOpenIm2d2Col.cl

```diff
@@ -212,37 +260,38 @@ kernel void Im2d2Col(const int data_size_off,
     inner_lid = lid;
     while(inner_lid < out_cols_wg * out_rows_wg)
     {
-        int out_x = inner_lid % out_cols_wg;
-        int out_y = inner_lid / out_cols_wg;
+        int out_x = inner_lid % out_cols_wg; /// < 256
+        int out_y = inner_lid / out_cols_wg; /// < 256
 
-        int col_x = (im_y + out_y) * out_w + im_x + out_x;
-        int col_y = (gid / NUM_IM_BLKS) * out_h * out_w * wei_h * wei_w;
+        index_t col_x = (index_t)(im_y + out_y) * out_w + im_x + out_x; /// out_h * out_w
+        /// c * out_h * out_w * wei_h * wei_w
+        index_t col_y = (gid / NUM_IM_BLKS) * out_h * out_w * wei_h * wei_w;
 
         for(int y = 0; y < wei_h; y++)
         {
             for(int x = 0; x < wei_w; x++)
             {
-                int im_off_h = out_y * stride_h + y * dilation_h;
-                int im_off_w = out_x * stride_w + x * dilation_w;
-                col[col_y + col_x + (y * wei_w + x) * out_h * out_w] =
-                    local_im[(im_off_h)*im_cols_wg + im_off_w];
+                int im_off_h    = out_y * stride_h + y * dilation_h;
+                int im_off_w    = out_x * stride_w + x * dilation_w;
+                index_t col_off = col_y + col_x + ((index_t)y * wei_w + x) * out_h * out_w;
+                col[col_off]    = local_im[(im_off_h)*im_cols_wg + im_off_w];
             }
         }
         inner_lid += 256;
     }
 #endif // NUM_IM_BLKS && STRIDE_GT_1
 #else
 
-    int tid = get_global_id(0);
-    while(tid < out_h * out_w * wei_w * wei_h * NUM_CH_TOTAL)
+    index_t tid = get_global_id(0);
+    while(tid < (index_t)out_h * out_w * wei_w * wei_h * NUM_CH_TOTAL)
     {
         // which row of the output to write to
-        int col_row = tid / (out_h * out_w);
+        index_t col_row = tid / ((index_t)out_h * out_w); // wei_w * wei_h * NUM_CH_TOTAL
 
         // which pixel from the image and which channel to read from
-        int im_x = col_row % wei_w;           // used to compute im_off_w
-        int im_y = (col_row / wei_w) % wei_h; // used to compute im_off_y
-        int im_c = col_row / (wei_w * wei_h); // im_c is the img channel
+        int im_x = col_row % wei_w;                     // used to compute im_off_w
+        int im_y = (col_row / wei_w) % wei_h;           // used to compute im_off_y
+        int im_c = (col_row / ((index_t)wei_w * wei_h); // im_c is the img channel
```

Comment:
```suggestion
        int im_c = (col_row / ((index_t)wei_w * wei_h)); // im_c is the img channel
```

Reply:
@shurale-nkn also I am testing on a Navi21 and get:
```
✗ ./bin/MIOpenDriver convfp16 -W 300 -H 100 -c 1024 -n 1 -k 4 -x 256 -y 256 -p 0 -q 0 -u 1 -v 1 -F 1 -t 1 -V 1 -l 2
MIOpenDriver convfp16 -W 300 -H 100 -c 1024 -n 1 -k 4 -x 256 -y 256 -p 0 -q 0 -u 1 -v 1 -F 1 -t 1 -V 1 -l 2
MIOpen(HIP): Info [get_device_name] Raw device name: gfx1030
MIOpen(HIP): Info [Handle] stream: 0x1045c00, device_id: 0
MIOpen(HIP): Info [GetFindModeValueImpl] MIOPEN_FIND_MODE = NORMAL(1)
MIOpen(HIP): Info [ForwardGetWorkSpaceSize] 
MIOpen(HIP): Info [AmdRocmMetadataVersionDetect] ROCm MD version AMDHSA_COv3, HIP version 5.2.22181, MIOpen version 2.18.0.9d4a9e6f7-dirty
MIOpen(HIP): Info [GetEnvFindOnlySolverImpl] 91
MIOpenDriver: /home/user/MIOpen/include/../test/tensor_holder.hpp:168: tensor<float>::tensor(miopen::TensorDescriptor) [T = float]: Assertion `rhs.GetType() == miopen_type<T>{} || ((miopen_type<T>{} == miopenInt8 || miopen_type<T>{} == miopenInt8x4) && rhs.GetType() == miopenFloat)' failed.
[1]    1249862 abort (core dumped)  ./bin/MIOpenDriver convfp16 -W 300 -H 100 -c 1024 -n 1 -k 4 -x 256 -y 256 -p

Reply:
hmm...
can you try with ?
export MIOPEN_FIND_MODE=1
export MIOPEN_DEBUG_FIND_ONLY_SOLVER=GemmFwdRest

Reply:
and please write the cmake options

Reply:
@shurale-nkn could you let me know which cmake option should be added?

Reply:
CXX=/opt/rocm/llvm/bin/clang++ CXXFLAGS='-Werror' cmake -DBUILD_DEV=On ..

Reply:
@shurale-nkn  yes I am using those, and still getting:
```
MIOpenDriver: /home/user/MIOpen/include/../test/tensor_holder.hpp:168: tensor<float>::tensor(miopen::TensorDescriptor) [T = float]: Assertion `rhs.GetType() == miopen_type<T>{} || ((miopen_type<T>{} == miopenInt8 || miopen_type<T>{} == miopenInt8x4) && rhs.GetType() == miopenFloat)' failed.

Reply:
@junliume lets try this

export MIOPEN_ENABLE_LOGGING=1
export MIOPEN_LOG_LEVEL=5

Reply:
@junliume I need logs. by mail

Reply:
@shurale-nkn I used log level 6 :)

Here is the complete log: 
[log.log](https://github.com/ROCmSoftwarePlatform/MIOpen/files/9194326/log.log)

Reply:
@atamazov  @JehandadKhan  it seems to be a host/API issue.

Reply:
update. @junliume 
This error occurs only if the project is created in debug mode (-DCMAKE_BUILD_TYPE=debug)
Maybe in your case debug is default.

Try to compile with:
CXX=/opt/rocm/llvm/bin/clang++ CXXFLAGS='-Werror' cmake -DBUILD_DEV=On -DCMAKE_BUILD_TYPE=release ..

Reply:
@junliume 
issue related to changes in PR [API] NCHWc tensor layout support #1532

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/d19c2575ca918f6b63a022f004d591795afcd85d/driver/conv_driver.hpp#L1278

https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1532/files#diff-2d7eb26c166e89e679bd68a41a7510ac3b8a66c1c2894e9dcf584e40711801ceL1232-L1238


before this PR was called tensor constructor from line:120
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/d19c2575ca918f6b63a022f004d591795afcd85d/test/tensor_holder.hpp#L120-L124

but now we are using constructor from line 164
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/d19c2575ca918f6b63a022f004d591795afcd85d/test/tensor_holder.hpp#L164-L170


so ```Tref=float``` => ```miopen_type<T> = miopen_type<float>```
but  ``(outputTensor type is half )`` => ```(rhs.GetType() = miopen_type<half_float::half>)```
which means 
```(  miopen_type<T> != rhs.GetType() )```

Reply:
Can you please help, if the test needs to be updated or is it a bug in MIOpen ?

Reply:
@JehandadKhan the issue mentioned above is tracked in #1653 and proposed fix is in #1657